.github/workflows/test.yaml

```diff
@@ -111,36 +126,59 @@ jobs:
   docker-lint:
     name: Dockerfile lint
     runs-on: ubuntu-22.04
+    outputs:
+      outcome: ${{ steps.report.outputs.outcome }}
+    defaults:
+      run:
+        working-directory: ${{ inputs.working-directory }}
     steps:
       - uses: actions/checkout@v3
       - name: Check for Dockerfiles
-        run: echo Dockerfiles=$(find . -name "*Dockerfile") >> $GITHUB_ENV
-      - if: ${{ env.Dockerfiles != '' }}
+        id: dockerfiles
+        run: echo "found=$(find . -name "*Dockerfile")" >> $GITHUB_OUTPUT
+      - if: ${{ steps.dockerfiles.outputs.found != '' }}
         name: Run HadoLint
         uses: jbergstroem/hadolint-gh-action@v1
         with:
-          dockerfile: "${{ env.Dockerfiles }}"
-  lint-metadata:
+          dockerfile: "${{ inputs.working-directory }}${{ steps.dockerfiles.outputs.found }}"
+      - name: Report
+        id: report
+        run: echo "outcome=$([ '${{ steps.dockerfiles.outputs.found }}' = '' ] && echo skip || echo success)" >> $GITHUB_OUTPUT
+  metadata-lint:
     name: Lint metadata.yaml
     runs-on: ubuntu-22.04
+    defaults:
+      run:
+        working-directory: ${{ inputs.working-directory }}
+    outputs:
+      outcome: ${{ steps.report.outputs.outcome }}
     steps:
       - uses: actions/checkout@v3
-      - name: Install check-jsonschema
-        run: python3 -m pip install check-jsonschema
-      - name: Download schema
-        run: curl -fLsSo $PWD/metadata.schema https://raw.githubusercontent.com/canonical/operator-workflows/main/.github/files/metadata.schema
+      - name: Check file existence
+        id: metadataYaml
+        run: echo "exists=$([ -f metadata.yaml ] && echo "true" || echo "false")" >> $GITHUB_OUTPUT
```

Comment:
Is is possible for a charm not to have a metadata.yaml file?

Reply:
No except we also use the operator workflow for the upload-charm-docs action